File: .agents/explorer_m1_2/verify_regex.js
const shareCode = "u2x1-3x2h0e10_51-1e15_3p4i5x3s1x2";
// Current regex in scripts/fetch-meta.ts
const currentRegex = /h([^\-dsu]+(?:-[^\-dsu]+)*)/;
// Proposed regex
const proposedRegex = /h([^\-dsui]+(?:-[^\-dsui]+)*)/;

const currentMatch = shareCode.match(currentRegex);
const proposedMatch = shareCode.match(proposedRegex);

console.log("Input Share Code:", shareCode);
console.log("Current Regex Match:", currentMatch ? currentMatch[0] : "null");
console.log("Current Captured Section:", currentMatch ? currentMatch[1] : "null");
console.log("Proposed Regex Match:", proposedMatch ? proposedMatch[0] : "null");
console.log("Proposed Captured Section:", proposedMatch ? proposedMatch[1] : "null");

// Let's test how they are parsed as hero strings:
if (currentMatch) {
  const heroesList = currentMatch[1].split('-');
  console.log("Current split heroes list:", heroesList);
  heroesList.forEach(heroStr => {
    const idMatch = heroStr.match(/^(\d+)/);
    const pMatch = heroStr.match(/p(\d+)/);
    const eMatch = heroStr.match(/e(\d+)(?:_(\d+))?/);
    console.log(`  HeroStr: "${heroStr}" -> idMatch: ${!!idMatch}, eMatch: ${!!eMatch}, pMatch: ${!!pMatch}`);
  });
}
if (proposedMatch) {
  const heroesList = proposedMatch[1].split('-');
  console.log("Proposed split heroes list:", heroesList);
  heroesList.forEach(heroStr => {
    const idMatch = heroStr.match(/^(\d+)/);
    const pMatch = heroStr.match(/p(\d+)/);
    const eMatch = heroStr.match(/e(\d+)(?:_(\d+))?/);
    console.log(`  HeroStr: "${heroStr}" -> idMatch: ${!!idMatch}, eMatch: ${!!eMatch}, pMatch: ${!!pMatch}`);
  });
}

File: .agents/explorer_m1_2/verify_regex.py
import re

share_code = "u2x1-3x2h0e10_51-1e15_3p4i5x3s1x2"

# Current regex
current_pattern = r'h([^\-dsu]+(?:-[^\-dsu]+)*)'
# Proposed regex
proposed_pattern = r'h([^\-dsui]+(?:-[^\-dsui]+)*)'

current_match = re.search(current_pattern, share_code)
proposed_match = re.search(proposed_pattern, share_code)

print("Input Share Code:", share_code)
if current_match:
    print("Current Regex Match:", current_match.group(0))
    print("Current Captured Section:", current_match.group(1))
    print("Current split heroes list:", current_match.group(1).split('-'))
else:
    print("Current Regex Match: None")

if proposed_match:
    print("Proposed Regex Match:", proposed_match.group(0))
    print("Proposed Captured Section:", proposed_match.group(1))
    print("Proposed split heroes list:", proposed_match.group(1).split('-'))
else:
    print("Proposed Regex Match: None")
